### PR TITLE
feat: add eNear connector package

### PR DESCRIPTION
**Pull request by paouvrard**
- Create eNEAR connector: `@near-eth/near-ether`
- Create `@near-eth/utils`, `find-replacement-tx` and `lite-merkle-patricia-tree` containing functionality common to all packages

https://enear.onrender.com/

### CODE REVIEW

File: .pnp.js [hunk 1]
```diff
@@ -3937,7 +3937,9 @@ function $$SETUP_STATE(hydrateRuntimeState, basePath) {
             ["@types/node", "npm:14.14.31"],
```
Comment (mfornet) [inline .pnp.js:3949]: what tool is generating this file?

Also, can we remove all .zip files introduced in this PR (or maybe the whole .yarn folder)?

File: packages/client/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@near-eth/client",
-  "version": "1.3.0-3",
+  "version": "1.3.0-5",
```
Comment (mfornet) [inline packages/client/package.json:3]: skip two versions o_O? is this because there is 1.3.0-4 in another PR?

File: packages/client/src/index.ts [hunk 3]
```diff
@@ -89,8 +89,7 @@ export function decorate (
 ): DecoratedTransfer {
   const type = getTransferType(transfer)
 
-  // not needed with structNullChecks false: @ts-expect-error
-  // to be improved
+  // @ts-expect-error
```
Comment (mfornet) [inline packages/client/src/index.ts:92]: why is this required here?

**Review by mfornet — APPROVED** [review]
Logic looks good to me.

Good job refactoring into typescript 👍 

Is this (https://github.com/aurora-is-near/rainbow-bridge-frontend/pull/233) the PR in the frontend that uses this new changes? I'll review the use of this changes in the frontend next.

File: .pnp.js (hunk 1 above)
Comment (paouvrard) [reply .pnp.js:3949]: `.php.js` and `.yarn` are generated by yarn2: they replace `node_modules` in a much more reliable way and since the cache is compressed, it is good practice to commit it so that yarn install doesn't need to download dependencies.

File: packages/client/package.json (hunk 2 above)
Comment (paouvrard) [reply packages/client/package.json:3]: I did some pre-preleases for testing (not all committed) which will be deleted and the release will be made from the `Main` branch following the yarn2 release process.

File: packages/client/src/index.ts (hunk 3 above)
Comment (paouvrard) [reply packages/client/src/index.ts:92]: Because `locale` is optional. There is another PR in draft (https://github.com/aurora-is-near/rainbow-bridge-client/pull/19) so I didn't want to modify this piece of logic in this PR and re-added `@ts-expect-error` since using strictNullChecks in tsconfig will cause `Type 'undefined' cannot be used as an index type.ts(2538)`.

**Comment by paouvrard** [comment]
> Is this (aurora-is-near/rainbow-bridge-frontend#233) the PR in the frontend that uses this new changes? I'll review the use of this changes in the frontend next.

yes
